components/style/values/specified/length.rs

```diff
@@ -220,14 +221,119 @@ impl CharacterWidth {
     }
 }
 
+/// Same as Gecko
+const ABSOLUTE_LENGTH_MAX: i32 = (1 << 30);
+const ABSOLUTE_LENGTH_MIN: i32 = - (1 << 30);
+
+/// Helper to convert a floating point length to application units
+fn to_au_round(length: CSSFloat, au_per_unit: CSSFloat) -> Au {
+    let au = (length * au_per_unit).round() as i32;
+    let clamped_au = min(max(au, ABSOLUTE_LENGTH_MIN), ABSOLUTE_LENGTH_MAX);
```

Comment:
No problem, I had doubts and should have voiced them :) Just pushed again. Here is the way I implemented it (without casting to f64 like before):

```rust
fn to_au_round(length: CSSFloat, au_per_unit: CSSFloat) -> Au {
    Au(
        (length * au_per_unit)
        .min(ABSOLUTE_LENGTH_MAX as f32)
        .max(ABSOLUTE_LENGTH_MIN as f32)
        .round() as i32
    )
}
```